tenzir/threatbus
# Use STIX-2 as internal format

###  :notebook_with_decorative_cover: Description

- Use [STIX-2 (version 2.1)](https://docs.oasis-open.org/cti/stix/v2.1/stix-v2.1.html) as internal transport format for [Indicators](https://docs.oasis-open.org/cti/stix/v2.1/cs02/stix-v2.1-cs02.html#_muftrcpnf89v) and [Sightings](https://docs.oasis-open.org/cti/stix/v2.1/cs02/stix-v2.1-cs02.html#_a795guqsap3r). This breaks all existing plugins.
- Migrate `in-memory` backbone plugin
- Migrate RabbitMQ backbone plugin 
- Migate `ZMQ-App` plugin
- The unit tests for all un-migrated plugins and `pyvast-threatbus` are disabled to make the CI happy. They will be migrated in separate PRs

###  :memo: Checklist


### :dart: Review Instructions



This PR can already be reviewed. Documentation will follow in the next iteration.

- Go commit-by-commit
- Use `tests/utils` to test all functionality

Walkthrough:
- Install a fresh virtualenv
- Start a local RabbitMQ (e.g., simply via [docker](https://hub.docker.com/_/rabbitmq))
- Install TB + only those plugins that changed in this PR (in-mem, rabbitmq, zmq)
- Start Threat Bus with the example config

Then you can use the `tests/utils/zmq_sender.py` and `tests/utils/zmq_receiver.py` to mimick small "apps". They connect to the ZMQ plugin and send messages back and forth. This should work independent of the backbone (in-mem or rabbit). Try both.
